corehq/apps/cloudcare/models.py

```diff
@@ -26,17 +26,15 @@ def save(self, force_insert=False, force_update=False, using=DEFAULT_DB_ALIAS, u
             force_insert=force_insert, force_update=force_update, using=using, update_fields=update_fields
         )
 
-    def user_can_access_app(self, user, app):
-        user_id = user['_id']
-        app_id = app['_id']
+    def user_can_access_app(self, user, app_id):
         if not self.restrict or user['doc_type'] == 'WebUser':
             return True
         app_group = None
         for app_group in self.sqlappgroup_set.all():
-            if app_group.app_id in (app_id, app['copy_of'] or ()):
+            if app_group.app_id == app_id:
                 break
         if app_group:
-            return Group.user_in_group(user_id, app_group.group_id)
+            return Group.user_in_group(user['_id'], app_group.group_id)
```

Comment:
This isn't related to your changes, but seems like there's a bug here where if there is no match, the loop ends without break, and `app_group` will incorrectly be the last iterated row

Reply:
That's right - I actually commented about that [on slack](https://dimagi.slack.com/archives/C02QG63MU/p1753457989042139).  I have the fix in a stash somewhere - meant to PR it separately but forgot.